[marathon] add queue gauge

jangie opened the pull request:
### What does this PR do?

This PR aims to add a queue gauge for marathon integration. This is helpful to see as it indicates that potentially there is a resource ask that is currently unsolvable, thus requiring either of notification or scaling out of resource availability.

### Motivation

Our current mesos/marathon setup does not have great insight into when we are running up against our resource limits for new application deployments. The aim with this is to allow us to, in future, automate scaling out resources.

### Testing Guidelines

To test this, set up a mesos/marathon cluster with this updated check code, and attempt to schedule a task that asks for more memory than is available on any mesos worker.

### Additional Notes

It would be extremely beneficial to give instruction on running tests locally for developers, as the README.md seems to be aimed more at check consumers than developers. This is thus a best effort with regards to testing. The added code for the check itself is working as expected on some of our test infrastructure.

masci commented on marathon/check.py:
```diff
@@ -74,6 +74,11 @@ def check(self, instance):
+        # Number of queued applications
+        response = self.get_json(urljoin(url, "v2/queue"), timeout, auth, acs_url, ssl_verify)
+        if response is not None:
+            self.gauge('marathon.queue', len(response['queue']), tags=instance_tags)
```
I see there are other info in the response payload, I'd name the metric `marathon.queue.size` so we leave room for future additions.

jangie commented:
@masci I believe I have addressed the comments, if there's anything remaining to resolve please let me know and I can get those going quickly. Thanks!

masci commented on marathon/metadata.csv:
```diff
@@ -8,3 +8,4 @@ marathon.instances,gauge,,,,Number of instances of this application to start,0,m
 marathon.mem,gauge,,mebibyte,,Memory that is needed for the application per instance,0,marathon,mem
 marathon.tasksRunning,gauge,,task,,Number of tasks running,0,marathon,tasks running
 marathon.tasksStaged,gauge,,task,,Number of tasks staged to run,0,marathon,tasks staged
+marathon.queue,gauge,,,,Number of apps waiting for deployment,0,marathon,apps
```
This has to be `marathon.queue.size` now

masci commented on marathon/manifest.json:
```diff
@@ -10,6 +10,6 @@
     "mac_os"
   ],
-  "version": "1.0.0",
+  "version": "1.0.1",
```
Since we're adding a new metric, can we do `1.1.0`?

jangie commented:
@masci thanks for the pointers! have addressed again.
